sdks/python/apache_beam/transforms/combiners.py

```diff
@@ -290,9 +291,12 @@ def __init__(self, n, compare=None, key=None, reverse=False):
   def _sort_buffer(self, buffer, lt):
     if lt in (operator.gt, operator.lt):
       buffer.sort(key=self._key_fn, reverse=self._reverse)
+    elif self._key_fn:
+      buffer.sort(key=cmp_to_key(
```

Comment:
Why can't we just use key=self._key_fn in this case?

Reply:
because you need both self._key_fn and and custom compare function lt in one key argument from the sort function

Reply:
In other words, we cannot use just `self._key_fn` because the order relationship is defined by `lt`, which is derived from `compare` - an external parameter to `TopCombineFn` and `Top` constructors. `self._key_fn` will return a key to use for comparisons, but the order will not always be captured but this key the constructors also accept an additional `compare` function.

Reply:
I think the intent was to accept one or the other, not to apply the key fn and then the comparator.

Reply:
Thanks for clarification, @robertwb. With this assumption, we can simplify the logic here. This assumption is also consistent with `Top` docstring, however I was confused by https://github.com/apache/beam/blob/1c573ffba014f6969a689d0e6cdcace31358885d/sdks/python/apache_beam/transforms/combiners.py#L319. 
robertwb@, would you be comfortable with merging PR as with a clean-up in a follow-up change? I can own the cleanup and can send you a PR with a few comment changes, and assertions here and there that ensure that that `key_fn` and `compare` are not used together.

Reply:
I didn't observe any performance changes with this PR.